api/http/offlinegate/offlinegate_wrapper.go

```diff
@@ -0,0 +1,58 @@
+package offlinegate
+
+import (
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+
+	httperror "github.com/portainer/libhttp/error"
+)
+
+const RedirectReasonAdminInitTimeout string = "AdminInitTimeout"
+
+// OfflineGateWrapper is an entity used to maintain the administrator initialization status
+type OfflineGateWrapper struct {
+	lock              *sync.Mutex
+	adminInitDisabled bool
+}
+
+// NewOfflineGateWrapper creates a new gate wrapper
+func NewOfflineGateWrapper(timeoutCh chan interface{}) *OfflineGateWrapper {
+	gateWrapper := &OfflineGateWrapper{
+		lock:              &sync.Mutex{},
```

Comment:
If not using a pointer, there is no need to initialize this.

Reply:
fixed